storage/s3_test.go

```diff
@@ -398,3 +404,150 @@ func TestS3Retry(t *testing.T) {
 		})
 	}
 }
+
+// Credit to aws-sdk-go
+func val(i interface{}, s string) interface{} {
+	v, err := awsutil.ValuesAtPath(i, s)
+	if err != nil || len(v) == 0 {
+		return nil
+	}
+	if _, ok := v[0].(io.Reader); ok {
+		return v[0]
+	}
+
+	if rv := reflect.ValueOf(v[0]); rv.Kind() == reflect.Ptr {
+		return rv.Elem().Interface()
+	}
+
+	return v[0]
+}
+
+func TestS3AclFlagOnCopy(t *testing.T) {
+	testcases := []struct {
+		name string
+		acl  string
+
+		expectedAcl string
+	}{
+		{
+			name: "no acl flag",
+		},
+		{
+			name: "acl flag without value, flag should be ignored",
```

Comment:
```suggestion
			name: "acl flag without value, flag should be ignored",
			acl:  "",
```

First two test cases seem identical.
Same for `put` tests.